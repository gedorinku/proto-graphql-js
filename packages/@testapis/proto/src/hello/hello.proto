syntax = "proto3";

package testapi.hello;

message Hello {
  // Required.
  Primitives required_primitives = 1;
  // Optional.
  Primitives optional_primitives = 2;
  // Required.
  repeated Primitives required_primitives_list = 3;
  // Optional.
  Primitives optional_primitives_list = 4;
}

message Primitives {
  double required_double_value =1;
  float required_float_value = 2;
  int32 required_int32_value = 3;
  int64 required_int64_value = 4;
  uint32 required_uint32_value = 5;
  uint64 required_uint64_value = 6;
  sint32 required_sint32_value = 7;
  sint64 required_sint64_value = 8;
  fixed32 required_fixed32_value = 9;
  fixed64 required_fixed64_value = 10;
  sfixed32 required_sfixed32_value = 11;
  sfixed64 required_sfixed64_value = 12;
  bool required_bool_value = 13;
  string required_string_value = 14;
  // bytes required_bytes_value = 1;
}
