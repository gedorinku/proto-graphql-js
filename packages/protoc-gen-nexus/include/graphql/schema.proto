syntax = "proto3";

package graphql;

import "google/protobuf/descriptor.proto";

message GraphqlSchemaOptions {
  string type_prefix = 1;
}

extend google.protobuf.FileOptions {
  GraphqlSchemaOptions schema = 2056;
}

message GraphqlFieldOptions {
  bool ignore = 1;
}

extend google.protobuf.FieldOptions {
  GraphqlFieldOptions field = 2056;
}

message GraphqlOneofOptions {
  bool ignore = 1;
}

extend google.protobuf.OneofOptions {
  GraphqlOneofOptions oneof = 2056;
}

message GraphqlEnumValueOptions {
  bool ignore = 1;
}

extend google.protobuf.EnumValueOptions {
  GraphqlEnumValueOptions enum_value = 2056;
}
