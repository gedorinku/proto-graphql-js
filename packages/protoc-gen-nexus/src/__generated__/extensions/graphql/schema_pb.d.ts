// package: graphql
// file: graphql/schema.proto

import * as jspb from "google-protobuf";
import * as google_protobuf_descriptor_pb from "google-protobuf/google/protobuf/descriptor_pb";

export class GraphqlSchemaOptions extends jspb.Message {
  getTypePrefix(): string;
  setTypePrefix(value: string): void;

  getIgnoreRequests(): boolean;
  setIgnoreRequests(value: boolean): void;

  getIgnoreResponses(): boolean;
  setIgnoreResponses(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlSchemaOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlSchemaOptions): GraphqlSchemaOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlSchemaOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlSchemaOptions;
  static deserializeBinaryFromReader(message: GraphqlSchemaOptions, reader: jspb.BinaryReader): GraphqlSchemaOptions;
}

export namespace GraphqlSchemaOptions {
  export type AsObject = {
    typePrefix: string,
    ignoreRequests: boolean,
    ignoreResponses: boolean,
  }
}

export class GraphqlObjectTypeOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  getSquashUnion(): boolean;
  setSquashUnion(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlObjectTypeOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlObjectTypeOptions): GraphqlObjectTypeOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlObjectTypeOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlObjectTypeOptions;
  static deserializeBinaryFromReader(message: GraphqlObjectTypeOptions, reader: jspb.BinaryReader): GraphqlObjectTypeOptions;
}

export namespace GraphqlObjectTypeOptions {
  export type AsObject = {
    ignore: boolean,
    squashUnion: boolean,
  }
}

export class GraphqlInputObjectTypeOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlInputObjectTypeOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlInputObjectTypeOptions): GraphqlInputObjectTypeOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlInputObjectTypeOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlInputObjectTypeOptions;
  static deserializeBinaryFromReader(message: GraphqlInputObjectTypeOptions, reader: jspb.BinaryReader): GraphqlInputObjectTypeOptions;
}

export namespace GraphqlInputObjectTypeOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlFieldOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlFieldOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlFieldOptions): GraphqlFieldOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlFieldOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlFieldOptions;
  static deserializeBinaryFromReader(message: GraphqlFieldOptions, reader: jspb.BinaryReader): GraphqlFieldOptions;
}

export namespace GraphqlFieldOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlOneofOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlOneofOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlOneofOptions): GraphqlOneofOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlOneofOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlOneofOptions;
  static deserializeBinaryFromReader(message: GraphqlOneofOptions, reader: jspb.BinaryReader): GraphqlOneofOptions;
}

export namespace GraphqlOneofOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlEnumOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlEnumOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlEnumOptions): GraphqlEnumOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlEnumOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlEnumOptions;
  static deserializeBinaryFromReader(message: GraphqlEnumOptions, reader: jspb.BinaryReader): GraphqlEnumOptions;
}

export namespace GraphqlEnumOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlEnumValueOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlEnumValueOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlEnumValueOptions): GraphqlEnumValueOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlEnumValueOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlEnumValueOptions;
  static deserializeBinaryFromReader(message: GraphqlEnumValueOptions, reader: jspb.BinaryReader): GraphqlEnumValueOptions;
}

export namespace GraphqlEnumValueOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

  export const schema: jspb.ExtensionFieldInfo<GraphqlSchemaOptions>;

  export const objectType: jspb.ExtensionFieldInfo<GraphqlObjectTypeOptions>;

  export const inputType: jspb.ExtensionFieldInfo<GraphqlInputObjectTypeOptions>;

  export const field: jspb.ExtensionFieldInfo<GraphqlFieldOptions>;

  export const oneof: jspb.ExtensionFieldInfo<GraphqlOneofOptions>;

  export const enumType: jspb.ExtensionFieldInfo<GraphqlEnumOptions>;

  export const enumValue: jspb.ExtensionFieldInfo<GraphqlEnumValueOptions>;

