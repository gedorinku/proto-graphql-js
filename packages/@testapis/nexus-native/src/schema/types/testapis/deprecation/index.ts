export * from "./deprecation_pb_nexus";
export * from "./file_deprecation_pb_nexus";
