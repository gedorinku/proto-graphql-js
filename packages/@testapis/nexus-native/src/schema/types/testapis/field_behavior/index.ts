export * from "./comments_pb_nexus";
