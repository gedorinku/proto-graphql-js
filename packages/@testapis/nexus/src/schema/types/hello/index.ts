export * from "./hello_pb_nexus";
