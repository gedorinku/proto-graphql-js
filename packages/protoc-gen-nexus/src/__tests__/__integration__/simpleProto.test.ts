import { graphql } from "graphql";
import { queryField } from "nexus";
import { testSchemaGeneration } from "../__helpers__/process.test.helper";
import * as pbjs from "@testapis/node/lib/testapis/hello";
import * as pbnative from "@testapis/node-native/lib/testapis/hello/hello_pb";

testSchemaGeneration("hello", "protobufjs", {
  schemaTests: [
    [
      "tests query",
      {
        test: queryField("test", {
          type: "Hello",
          resolve() {
            return new pbjs.testapis.hello.Hello({
              requiredPrimitives: new pbjs.testapis.hello.Primitives({
                requiredDoubleValue: 2.4,
                requiredFloatValue: 3.5,
                requiredInt32Value: 2,
                requiredInt64Value: 4,
                requiredUint32Value: 5,
                requiredUint64Value: 6,
                requiredSint32Value: 7,
                requiredSint64Value: 8,
                requiredFixed32Value: 9,
                requiredFixed64Value: 10,
                requiredSfixed32Value: 11,
                requiredSfixed64Value: 12,
                requiredBoolValue: true,
                requiredStringValue: "foobar",
              }),
              requiredPrimitivesList: [],
            });
          },
        }),
      },
      async (schema) => {
        expect(
          await graphql(
            schema,
            `
              query Test {
                test {
                  requiredPrimitives {
                    ...Primitives
                  }
                  optionalPrimitives {
                    ...Primitives
                  }
                  requiredPrimitivesList {
                    ...Primitives
                  }
                  optionalPrimitivesList {
                    ...Primitives
                  }
                }
              }
              fragment Primitives on Primitives {
                requiredDoubleValue
                requiredFloatValue
                requiredInt32Value
                requiredInt64Value
                requiredUint32Value
                requiredUint64Value
                requiredSint32Value
                requiredSint64Value
                requiredFixed32Value
                requiredFixed64Value
                requiredSfixed32Value
                requiredSfixed64Value
                requiredBoolValue
                requiredStringValue
              }
            `
          )
        ).toMatchSnapshot();
      },
    ],
  ],
});

testSchemaGeneration("hello", "native protobuf", {
  schemaTests: [
    [
      "tests query",
      {
        test: queryField("test", {
          type: "Hello",
          resolve() {
            const rp = new pbnative.Primitives();
            rp.setRequiredDoubleValue(2.4);
            rp.setRequiredFloatValue(3.5);
            rp.setRequiredInt32Value(2);
            rp.setRequiredInt64Value(4);
            rp.setRequiredUint32Value(5);
            rp.setRequiredUint64Value(6);
            rp.setRequiredSint32Value(7);
            rp.setRequiredSint64Value(8);
            rp.setRequiredFixed32Value(9);
            rp.setRequiredFixed64Value(10);
            rp.setRequiredSfixed32Value(11);
            rp.setRequiredSfixed64Value(12);
            rp.setRequiredBoolValue(true);
            rp.setRequiredStringValue("foobar");
            const hello = new pbnative.Hello();
            hello.setRequiredPrimitives(rp);
            return hello;
          },
        }),
      },
      async (schema) => {
        expect(
          await graphql(
            schema,
            `
              query Test {
                test {
                  requiredPrimitives {
                    ...Primitives
                  }
                  optionalPrimitives {
                    ...Primitives
                  }
                  requiredPrimitivesList {
                    ...Primitives
                  }
                  optionalPrimitivesList {
                    ...Primitives
                  }
                }
              }
              fragment Primitives on Primitives {
                requiredDoubleValue
                requiredFloatValue
                requiredInt32Value
                requiredInt64Value
                requiredUint32Value
                requiredUint64Value
                requiredSint32Value
                requiredSint64Value
                requiredFixed32Value
                requiredFixed64Value
                requiredSfixed32Value
                requiredSfixed64Value
                requiredBoolValue
                requiredStringValue
              }
            `
          )
        ).toMatchSnapshot();
      },
    ],
  ],
});
