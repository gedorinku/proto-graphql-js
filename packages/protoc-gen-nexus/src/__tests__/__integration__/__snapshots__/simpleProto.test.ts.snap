// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with native protobuf can make GraphQL schema from generated DSLs: schema.graphql 1`] = `
"### This file was generated by Nexus Schema
### Do not make changes to this file directly


\\"\\"\\"\\"\\"\\"
type Hello {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: Primitives

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: Primitives

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: Primitives!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [Primitives!]
}

\\"\\"\\"\\"\\"\\"
input HelloInput {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: PrimitivesInput

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: PrimitivesInput

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: PrimitivesInput!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [PrimitivesInput!]
}

\\"\\"\\"\\"\\"\\"
type Primitives {
  \\"\\"\\"\\"\\"\\"
  requiredBoolValue: Boolean!

  \\"\\"\\"\\"\\"\\"
  requiredDoubleValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredFixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredFixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredFloatValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredInt32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredInt64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSint64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredStringValue: String!

  \\"\\"\\"\\"\\"\\"
  requiredUint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredUint64Value: String!
}

\\"\\"\\"\\"\\"\\"
input PrimitivesInput {
  \\"\\"\\"\\"\\"\\"
  requiredBoolValue: Boolean!

  \\"\\"\\"\\"\\"\\"
  requiredDoubleValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredFixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredFixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredFloatValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredInt32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredInt64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSint64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredStringValue: String!

  \\"\\"\\"\\"\\"\\"
  requiredUint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredUint64Value: String!
}

type Query {
  ok: Boolean!
}
"
`;

exports[`schema generation with native protobuf tests query 1`] = `
Object {
  "data": Object {
    "test": Object {
      "optionalPrimitives": null,
      "optionalPrimitivesList": null,
      "requiredPrimitives": Object {
        "requiredBoolValue": true,
        "requiredDoubleValue": 2.4,
        "requiredFixed32Value": 9,
        "requiredFixed64Value": "10",
        "requiredFloatValue": 3.5,
        "requiredInt32Value": 2,
        "requiredInt64Value": "4",
        "requiredSfixed32Value": 11,
        "requiredSfixed64Value": "12",
        "requiredSint32Value": 7,
        "requiredSint64Value": "8",
        "requiredStringValue": "foobar",
        "requiredUint32Value": 5,
        "requiredUint64Value": "6",
      },
      "requiredPrimitivesList": Array [],
    },
  },
}
`;

exports[`schema generation with protobufjs can make GraphQL schema from generated DSLs: schema.graphql 1`] = `
"### This file was generated by Nexus Schema
### Do not make changes to this file directly


\\"\\"\\"\\"\\"\\"
type Hello {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: Primitives

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: Primitives

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: Primitives!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [Primitives!]
}

\\"\\"\\"\\"\\"\\"
input HelloInput {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitives: PrimitivesInput

  \\"\\"\\"Optional.\\"\\"\\"
  optionalPrimitivesList: PrimitivesInput

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitives: PrimitivesInput!

  \\"\\"\\"Required.\\"\\"\\"
  requiredPrimitivesList: [PrimitivesInput!]
}

\\"\\"\\"\\"\\"\\"
type Primitives {
  \\"\\"\\"\\"\\"\\"
  requiredBoolValue: Boolean!

  \\"\\"\\"\\"\\"\\"
  requiredDoubleValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredFixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredFixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredFloatValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredInt32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredInt64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSint64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredStringValue: String!

  \\"\\"\\"\\"\\"\\"
  requiredUint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredUint64Value: String!
}

\\"\\"\\"\\"\\"\\"
input PrimitivesInput {
  \\"\\"\\"\\"\\"\\"
  requiredBoolValue: Boolean!

  \\"\\"\\"\\"\\"\\"
  requiredDoubleValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredFixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredFixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredFloatValue: Float!

  \\"\\"\\"\\"\\"\\"
  requiredInt32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredInt64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSfixed64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredSint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredSint64Value: String!

  \\"\\"\\"\\"\\"\\"
  requiredStringValue: String!

  \\"\\"\\"\\"\\"\\"
  requiredUint32Value: Int!

  \\"\\"\\"\\"\\"\\"
  requiredUint64Value: String!
}

type Query {
  ok: Boolean!
}
"
`;

exports[`schema generation with protobufjs tests query 1`] = `
Object {
  "data": Object {
    "test": Object {
      "optionalPrimitives": null,
      "optionalPrimitivesList": null,
      "requiredPrimitives": Object {
        "requiredBoolValue": true,
        "requiredDoubleValue": 2.4,
        "requiredFixed32Value": 9,
        "requiredFixed64Value": "10",
        "requiredFloatValue": 3.5,
        "requiredInt32Value": 2,
        "requiredInt64Value": "4",
        "requiredSfixed32Value": 11,
        "requiredSfixed64Value": "12",
        "requiredSint32Value": 7,
        "requiredSint64Value": "8",
        "requiredStringValue": "foobar",
        "requiredUint32Value": 5,
        "requiredUint64Value": "6",
      },
      "requiredPrimitivesList": Array [],
    },
  },
}
`;
